Update MAPL to v2.12.3

MAPL v2.11.0 does not build on M1 Mac, complaining about `Unknown processor`. This should be fixed with a newer version of MAPL. A request for MAPL v2.12.3 was made, so let's do it here too.

https://github.com/NOAA-EMC/hpc-stack/issues/460

Same for MAPL 2.12.3:
```
/work/spack-stack/spack-stack-package-cleanup-20220602/envs/jedi-all-apple-clang-openmpi-try3/install/apple-clang/13.1.6/openmpi-4.1.3-hldiips/bin/mpic++' '-DCMAKE_Fortran_COMPILER=/Users/heinzell/work/spack-stack/spack-stack-package-cleanup-20220602/envs/jedi-all-apple-clang-openmpi-try3/install/apple-clang/13.1.6/openmpi-4.1.3-hldiips/bin/mpif90' '-DCMAKE_Fortran_FLAGS:STRING=-ffree-line-length-none -fallow-invalid-boz -fallow-argument-mismatch' '/Users/heinzell/work/spack-stack/spack-stack-package-cleanup-20220602/cache/build_stage/spack-stage-mapl-2.12.3-fbzwdve5hlw3nbgrnmbahj5cqie6rn3x/spack-src'

1 error found in build log:
     78    Call Stack (most recent call first):
     79      ESMA_cmake/GNU.cmake:77 (ecbuild_warn)
     80      ESMA_cmake/esma.cmake:28 (include)
     81      CMakeLists.txt:29 (include)
     82
     83
  >> 84    CMake Error at ESMA_cmake/GNU.cmake:127 (message):
     85      Unknown processor.  Contact Matt Thompson
     86    Call Stack (most recent call first):
     87      ESMA_cmake/esma.cmake:28 (include)
     88      CMakeLists.txt:29 (include)
     89
     90

See build log for details:
  /Users/heinzell/work/spack-stack/spack-stack-package-cleanup-20220602/cache/build_stage/spack-stage-mapl-2.12.3-fbzwdve5hlw3nbgrnmbahj5cqie6rn3x/spack-build-out.txt
```
Can we revert back to the old (2.8.x version on macOS)? Who needs to fix that? Are they aware?

--- a/.gitmodules
+++ b/.gitmodules
@@ -1,9 +1,9 @@
 [submodule "spack"]
   path = spack
-  #url = https://github.com/spack/spack
-  #branch = develop
-  url = https://github.com/NOAA-EMC/spack
-  branch = jcsda_emc_spack_stack
+  url = https://github.com/kgerheiser/spack.git
+  branch = mapl-fixes
+  #url = https://github.com/NOAA-EMC/spack
+  #branch = jcsda_emc_spack_stack
 [submodule "doc/CMakeModules"]
   path = doc/CMakeModules
   url = https://github.com/noaa-emc/cmakemodules

--- a/.gitmodules
+++ b/.gitmodules
@@ -1,9 +1,9 @@
 [submodule "spack"]
   path = spack
-  url = https://github.com/kgerheiser/spack.git
-  branch = mapl-fixes
-  #url = https://github.com/NOAA-EMC/spack
-  #branch = jcsda_emc_spack_stack
+  #url = https://github.com/spack/spack
+  #branch = develop
+  url = https://github.com/NOAA-EMC/spack
+  branch = jcsda_emc_spack_stack
 [submodule "doc/CMakeModules"]
   path = doc/CMakeModules
   url = https://github.com/noaa-emc/cmakemodules

--- a/configs/common/packages.yaml
+++ b/configs/common/packages.yaml
@@ -116,7 +116,7 @@
     grib-util:
       version: [1.2.3]
     gftl-shared:
-      version: [1.3.3]
+      version: [1.3.6]
     jedi-cmake:
       version: [1.3.0]
     # Attention - when updating also check orion site config
@@ -235,6 +235,6 @@
     metplus:
       version: [4.1.0]
     mapl:
-      version: [2.11.0]
+      version: [2.12.3]
     yafyaml:
       version: [0.5.1]

--- a/configs/common/packages.yaml
+++ b/configs/common/packages.yaml
@@ -116,7 +116,7 @@
     grib-util:
       version: [1.2.3]
     gftl-shared:
-      version: [1.3.6]
+      version: [1.5.0]
     jedi-cmake:
       version: [1.3.0]
     # Attention - when updating also check orion site config

